applications/RANSApplication/custom_processes/rans_line_output_process.cpp

```diff
@@ -0,0 +1,516 @@
+    const auto& r_process_info = r_model_part.GetProcessInfo();
+
+    if (KratosComponents<Variable<int>>::Has(mOutputStepControlVariableName)) {
+        const auto& r_variable = KratosComponents<Variable<int>>::Get(mOutputStepControlVariableName);
+        KRATOS_ERROR_IF(!r_process_info.Has(r_variable))
+            << "\"output_step_control_variable_name\" ( " << mOutputStepControlVariableName
+            << " ) not found in process info of " << r_model_part.Name() << ".\n";
+    } else if (KratosComponents<Variable<double>>::Has(mOutputStepControlVariableName)) {
+        const auto& r_variable = KratosComponents<Variable<double>>::Get(mOutputStepControlVariableName);
+        KRATOS_ERROR_IF(!r_process_info.Has(r_variable))
+            << "\"output_step_control_variable_name\" ( " << mOutputStepControlVariableName
+            << " ) not found in process info of " << r_model_part.Name() << ".\n";
+    } else if (KratosComponents<Variable<std::string>>::Has(mOutputStepControlVariableName)) {
+        const auto& r_variable = KratosComponents<Variable<std::string>>::Get(mOutputStepControlVariableName);
+        KRATOS_ERROR_IF(!r_process_info.Has(r_variable))
+            << "\"output_step_control_variable_name\" ( " << mOutputStepControlVariableName
+            << " ) not found in process info of " << r_model_part.Name() << ".\n";
+    }
+
+    const auto& r_direction_vector = mEndPoint - mStartPoint;
+
+    mSamplingPoints.resize(mNumberOfSamplingPoints * 3);
+    mSamplingPointElementIds.resize(mNumberOfSamplingPoints, 0);
+    mSamplingPointElementShapeFunctions.resize(mNumberOfSamplingPoints);
+
+    const auto& r_communicator = r_model_part.GetCommunicator().GetDataCommunicator();
+
+    // calculate the sampling points in the rank zero
+    if (r_communicator.Rank() == 0) {
+        unsigned int local_index = 0;
+        for (int i = 0; i < mNumberOfSamplingPoints; ++i) {
+            const auto& current_point =
+                mStartPoint + r_direction_vector *
+                                  (static_cast<double>(i) /
+                                   static_cast<double>(mNumberOfSamplingPoints - 1));
+            mSamplingPoints[local_index++] = current_point[0];
+            mSamplingPoints[local_index++] = current_point[1];
+            mSamplingPoints[local_index++] = current_point[2];
+        }
+    }
+
+    r_communicator.Broadcast(mSamplingPoints, 0);
+
+    BruteForcePointLocator brute_force_point_locator(r_model_part);
+    for (int i = 0; i < mNumberOfSamplingPoints; ++i) {
+        Point current_point(mSamplingPoints[i * 3], mSamplingPoints[i * 3 + 1],
+                            mSamplingPoints[i * 3 + 2]);
+        mSamplingPointElementIds[i] = brute_force_point_locator.FindElement(
+            current_point, mSamplingPointElementShapeFunctions[i]);
+        if (mSamplingPointElementIds[i] > -1) {
+            mSamplePointLocalIndexList.push_back(i);
+        }
```

Comment:
Maybe it's a good idea to throw a warning for those points that are not found. However this might be too verbose in those cases in which the line endpoints are outside the computational domain. We can introduce an echo level for these cases.

Reply:
(now I see that this it's checked below after gathering the values)